client/core/core.go

```diff
@@ -2083,6 +2122,77 @@ func handleRevokeMatchMsg(_ *Core, dc *dexConnection, msg *msgjson.Message) erro
 	return tracker.db.UpdateOrder(metaOrder)
 }
 
+// handleTradeSuspensionMsg is called when a trade suspension notification is received.
+func handleTradeSuspensionMsg(c *Core, dc *dexConnection, msg *msgjson.Message) error {
+	var suspension msgjson.TradeSuspension
+	err := msg.Unmarshal(&suspension)
+	if err != nil {
+		return fmt.Errorf("trade suspension unmarshal error: %v", err)
+	}
+
+	dc.suspensionsMtx.Lock()
+	dc.suspensions[suspension.MarketID] = &suspension
+	dc.suspensionsMtx.Unlock()
+
+	go func(c *Core, dc *dexConnection, suspension *msgjson.TradeSuspension) {
+		nowMilli := encode.UnixMilliU(time.Now())
+
+		var duration time.Duration
+		if nowMilli < suspension.SuspendTime {
+			milliDiff := suspension.SuspendTime - nowMilli
+			duration = time.Duration(milliDiff * 1e6)
+		}
+
+		timer := time.NewTimer(duration)
+		for {
+			select {
+			case <-c.ctx.Done():
+				return
+
+			case <-timer.C:
+				dc.booksMtx.Lock()
+				defer dc.booksMtx.Unlock()
+
+				bookie, ok := dc.books[suspension.MarketID]
+				if !ok {
+					log.Errorf("no bookie found with market id %s",
+						suspension.MarketID)
+				}
+
+				switch suspension.Persist {
+				case true:
+					bookie.Reset()
+
+				case false:
+					dc.tradeMtx.Lock()
+					toDelete := make([]*trackedTrade, 0)
+					for _, trade := range dc.trades {
```

Comment:
The trades aren't stopped on market suspend, it's just that the booked orders are revoked.  Active trades for revoked orders continue negotiation.

I imagine you'd use `db.DB.MarketOrders` or `db.DB.ActiveDEXOrders` or a new one like `MarketOrders` that also filters by status to get the orders for this dex-market, and put them in revoked status.